client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import noop from 'lodash/noop';
+import i18n from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Spinner from 'components/spinner';
+import { getCurrentUserEmail } from 'state/current-user/selectors';
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( this.props.emailSent || this.props.error ) {
+			return i18n.translate( 'Email Sent' );
+		}
+		if ( this.props.pendingRequest ) {
+			return <Spinner className="email-verification-dialog__confirmation-dialog-spinner" />;
+		}
+		return i18n.translate( 'Resend Email' );
+	}
+
+	handleClose = () => {
+		this.props.resetVerifyEmailState();
+		this.props.onClose();
+	};
+
+	getDialogButtons() {
+		return [
+			<Button
+				key="resend"
+				primary={ false }
+				disabled={ this.props.pendingRequest || this.props.emailSent || this.props.error }
+				onClick={ this.props.verifyEmail }>
+				{ this.getResendButtonLabel() }
+			</Button>,
+			<Button
+				key="close"
+				primary={ true }
+				onClick={ this.handleClose }>
+					{ i18n.translate( 'OK' ) }
+			</Button>
+		];
+	}
+
+	render() {
+		const strings = {
+			confirmHeading: i18n.translate( 'Please confirm your email address' ),
+
+			confirmExplanation: i18n.translate( 'We sent you an email when you first signed up. ' +
+				'Please open the message and click the blue button.' ),
+
+			confirmReasoning: i18n.translate( 'Email confirmation allows us to assist when recovering ' +
+				'your account in the event you forget your password.' ),
+
+			confirmEmail: i18n.translate(
```

Comment:
Please note that we usually wrap a component with `localize` and use `this.props.translate()` instead of calling to `i18n` directly.

```js
import { localize } from 'i18n-calypso';

export Thing = ( { id, translate } ) => <Stuff id={ id } name={ translate( 'Bob' ) } />

export default localize( Thing );
```